validator/project_validator.go

```diff
@@ -1898,3 +1899,37 @@ func parseS3PullParameters(c model.PluginCommandConf) (task, bv string, err erro
 	}
 	return task, bv, nil
 }
+
+// checkTaskTimeout checks if all tasks contain a timeout
+func checkTaskTimeout(project *model.Project) ValidationErrors {
+	errs := ValidationErrors{}
+	if project.ExecTimeoutSecs > 0 {
+		return errs
+	}
+	taskNames := []string{}
+	for _, task := range project.Tasks {
+		if task.ExecTimeoutSecs == 0 {
+			taskNames = append(taskNames, task.Name)
+		}
+	}
+	if len(taskNames) == 1 {
+		errs = append(errs,
+			ValidationError{
+				Message: fmt.Sprintf("task '%s' in project '%s' does not "+
+					"have an exec_timeout_secs defined",
+					taskNames[0], project.Identifier),
+				Level: Warning,
+			},
+		)
+	} else if len(taskNames) > 1 {
+		errs = append(errs,
+			ValidationError{
+				Message: fmt.Sprintf("tasks '%s' in project '%s' do not "+
+					"have an exec_timeout_secs defined",
+					strings.Join(taskNames, ", "), project.Identifier),
```

Comment:
There are some projects that might have this warning true for many many tasks in which case this could be a super long warning message - can you instead follow something like the message at this line? https://github.com/evergreen-ci/evergreen/blob/main/validator/project_validator.go#L567 Where we pick up if one or more tasks fit the warning criteria and then simply remind the user that tasks must have an exec_timeout_secs defined for them.  Can you also include part of the message to say what default timeout the tasks will fall back to if nothing is defined? (I believe its like 4 hours but not sure).